erpnext/public/js/controllers/buying.js

```diff
@@ -213,6 +212,22 @@ erpnext.buying.BuyingController = erpnext.TransactionController.extend({
 		this.qty(doc, cdt, cdn);
 	},
 
+	rate: function(doc,cdt,cdn) {
+		this.calculate_stock_uom_rate(doc, cdt, cdn);
+		this._super(doc, cdt, cdn);
+	},
+
+	conversion_factor: function(doc,cdt,cdn) {
+		this.calculate_stock_uom_rate(doc, cdt, cdn);
+		this._super(doc, cdt, cdn);
+	},
+
+	calculate_stock_uom_rate: function(doc, cdt, cdn) {
+		let item = frappe.get_doc(cdt, cdn);
+		item.stock_uom_rate = item.rate/item.conversion_factor;
```

Comment:
```suggestion
		item.stock_uom_rate = flt(item.rate) / (flt(item.conversion_factor) || 1);
```
Better to be cautious